### ISSUE TITLE
why batch size pre device not work

### ISSUE THREAD

**Issue by Link-Li** [huggingface/trl#3165]
I have a training set of 800,000 samples. Based on DeepSpeed ZeRO-1, I am using 24 GPUs for training and 8 GPUs for VLLM, and I am using GRPO for training. I set per_device_train_batch_size to 3, gradient_accumulation_steps to 1, and epoch to 1. However, I found that there are 800,000 optimization steps, which I don't quite understand. Shouldn't it be 800,000 / 3 / 24 = 11,112 steps?

here is training print log：

***** Running training *****
  Num examples = 800,000
  Num Epochs = 1
  Instantaneous batch size per device = 3
  Total train batch size (w. parallel, distributed & accumulation) = 72
  Gradient Accumulation steps = 1
  Total optimization steps = 800,000
  Number of trainable parameters = 8,809,353,216

**Comment by qgallouedec** [comment]
You need to multiply by the number of generations per prompt (8 by default)
